azure-mgmt-compute/src/main/java/com/azure/management/compute/implementation/VirtualMachineScaleSetImpl.java

```diff
@@ -1902,17 +1906,18 @@ private void loadCurrentPrimaryLoadBalancersIfAvailable() throws IOException {
         }
 
         if (firstLoadBalancerId == null) {
-            return;
+            return self;
         }
 
-        LoadBalancer loadBalancer1 = this.networkManager
-                .loadBalancers()
-                .getById(firstLoadBalancerId);
-        if (loadBalancer1.publicIPAddressIds() != null && loadBalancer1.publicIPAddressIds().size() > 0) {
-            this.primaryInternetFacingLoadBalancer = loadBalancer1;
-        } else {
-            this.primaryInternalLoadBalancer = loadBalancer1;
-        }
+        self = self.concatWith(Mono.just(firstLoadBalancerId).flatMap(id -> this.networkManager.loadBalancers().getByIdAsync(id)
```

Comment:
`self.map(ignored -> firstLoadBalancerId).flatMap(id -> ...` might be better, since it keeps `Mono` (if I did not misunderstand, self is `Mono.just(this)`)